core/src/main/java/org/primefaces/extensions/model/mongo/MorphiaLazyDataModel.java

```diff
@@ -248,6 +251,61 @@ public int count(final Map<String, FilterMeta> map) {
                                 }
 
                                 break;
+                                
+                            case BETWEEN:
+                                if (metadata.getFilterValue().getClass() == ArrayList.class) {
+                                    final List dates = (ArrayList) metadata.getFilterValue();
+                                    if (dates.size() > 1){ //does this ever have less than 2 items?
+                                        q.filter(Filters.gt(field, dates.get(0)),Filters.lt(field,dates.get(1)));
```

Comment:
Shouldn't this be `gte`?
```suggestion
                                        q.filter(Filters.gte(field, dates.get(0)), Filters.lt(field,dates.get(1)));
```

Reply:
At first I was using gte and lte to include both the start and end date, but I wasn't sure if we should  make the assumption that between queries include the start and end date, so I chose to just use lt and gt and not include them.

I dont have a strong preference one way or another, should between be date inclusive?

Reply:
Maybe apples and pears, but in ANSI SQL they are both inclusive. But, with dates you might want to exclude the end date as explained here https://stackoverflow.com/questions/749615/does-ms-sql-servers-between-include-the-range-boundaries/749663

Reply:
Hmm yea in mongo there is no between keyword, "between queries" are usually done with lt, lte, gt, gte. That said, I was now looking at the "between" match mode for non-lazy datatables on the showcase here: https://www.primefaces.org/showcase/ui/data/datatable/filter.xhtml?jfwid=abd89

And I noticed that it includes the start and end date, so maybe we should match that and make this query include the dates aswell? What do you think?

Reply:
Yes, I would match the showcase, so both inclusive. @melloware WDYT?

Maybe we should break up each mode into separate methods, so they are easily overridable. So, you can extend this class and only override the between filter method (or whatever method you want to change).

Reply:
Yes we should match the showcase and PF original intent so if that is what it's doing it's what we should do too.

Reply:
That sounds good too.

I had an idea the other day which I was going to propose, although it's not PFE related, it depends on PF, but since we're on the subject... 

It would be great if we allowed the user to specify a custom match mode. At the moment if you specify a match mode that doesn't exist, the filterMeta matchmode returns null. It would be nice if it returned something like "CUSTOM" and if we had a way to then access the matchmode name specified on the p:column's filterMatchMode="". That way, the generic data models like this one and JPA's could have a method like .matchMode("customMatchModeName", BiConsumer< Query, FilterMeta > which would allow the user to implement their custom match mode.

That said, I'm not familiar with PF's internal matchmode implementation and dont know if the way the current structure is set up would allow for such a change.

Reply:
PrimeReact has a CUSTOM so I don't see why we cant add it to PF

Reply:
Alright I changed it so between is date inclusive and applied the spotless formatter, I think the build is no longer failing. I dont think I'll have much free time for the next few days so I'm not sure when I'll get to split up each match mode into separate methods, but maybe we can leave that for a future PR if you all agree